lib/cocoapods/target/build_settings.rb

```diff
@@ -538,8 +542,7 @@ def initialize(target, test_spec = nil)
 
         # @return [Array<String>]
         define_build_settings_method :weak_frameworks, :memoized => true do
-          return [] if (!target.requires_frameworks? || target.static_framework?) && !test_xcconfig?
-
+          return [] unless test_xcconfig?
```

Comment:
that doesnt look right

Reply:
well wait it might be

Reply:
Its correct https://github.com/CocoaPods/CocoaPods/pull/8314/files#diff-54f9c07edbb5b7fc64e04683656d1778R565

Reply:
previous logic was a bit odd since `static_framework?` anyway

Reply:
@paulb777 does this mean this is OK?

Reply:
Yes, it's ok.  I did a self-correcting loop. :)